Fix default options being fully overridden by props.options in widget components

aligu7 opened the pull request:
### 🐛 Problem

Currently, all TradingView widget components use the `props.options || defaults` pattern, which **completely overrides** the default options if any `props.options` are passed. This makes it impossible to only override specific fields (e.g. just `symbol`) while preserving the rest of the defaults.

Example:
```vue
<TvChart :options="{ symbol: 'BINANCE:BTCUSDT' }" />

aligu7 committed:
fix: allow merging user-defined options with default widget options

volkanakkus commented:
@aligu7, thanks for the contribution. About the issue, you're totally right, but to update the module, we can use withDefaults() for this feature without needing lodash. 

From Nuxt 3, since it’s built on Vue 3.2+, we can easily set default prop values and override only specific fields in a clean and type-safe way. Here’s a example:

```ts
const props = withDefaults(defineProps<Props>(), {
  symbol: 'BTCUSDT', // default symbol
  // other defaults...
})

```
This way, we can preserve all default options and only override what’s needed. 
May we can update the PR like that, what do you think?

aligu7 committed:
fix: merge default options with user-defined options using withDefaults

aligu7 commented:
Thanks for the feedback @volkanakkus
You're totally right, using `withDefaults` is a cleaner and more Nuxt-native way to handle this.

I've removed the previous `lodash.merge` approach and updated the PR to use `withDefaults` + `Partial<T>` props, merging them into default options in a type-safe way. This keeps the override behavior flexible while preserving all defaults.

Let me know if anything else needs adjusting :)

aligu7 commented:
@volkanakkus, also a lot of modules in the project in the `package.json` are outdated and needs to be updated. Locally I tried updating all of them to latest version. All of them worked, except `eslint` requires config file addition (`eslint.config.(js|mjs|cjs)`), which if you want I can add easily. Without this it is not possible to lint files, unless he old version of the eslint will be used, which I don't find practical. But, besides that, this PR is ready to be merged.

volkanakkus commented:
@aligu7 I’ll handle the dependency bumps in a separate PR to keep things tidy.

But nice work here! 🙌

There's just one small thing I noticed:
Since you're typing `options` as Partial<ChartOptions>, TypeScript will complain if someone wants to pass a new TradingView parameter that's not declared in ChartOptions. Why I'm saying this, to support fully typed options of TradingView parameters is so impossible, with every change of theirs, we need to update the module again and again.  

To allow for future props without losing IntelliSense for existing ones, I’d suggest changing it to:

```ts
options?: Partial<ChartOptions> & { [key: string]: unknown }
```

That way, known keys are fully typed, but adding new/custom TradingView config keys won’t break anything.

aligu7 committed:
fix: support custom TradingView options with extended typing

aligu7 commented:
Thanks! Just updated the options typing as suggested to allow custom TradingView keys while keeping IntelliSense.

volkanakkus commented:
Okay thanks, I'm going to review & test the changes, and I'll let you know when I'm done

volkanakkus commented:
You know what, this whole PR should stand on an optional value from `nuxt.config.ts` because;

Some users might be passing only their own custom values for certain widgets, like stocks or others, without relying on the default options at all. 

If we force merging defaults with their configs in this new version, they could suddenly get unexpected extra default values merged in, which might unintentionally change how their widget behaves or looks, potentially breaking their existing setups. 

Also, if we don’t make this optional and force default merging for everyone, this would actually be a **_breaking change_**, because it alters existing behavior and could cause issues for current users. So to keep backward compatibility and avoid classifying this update as breaking, it makes sense to keep this merging behavior as an optional setting that users can enable when needed.

So, I'm going to make this feature as optional and let you know.

Commit:
fix: move the defaultWidgetOptions to data folder

aligu7 commented:
That makes total sense. Making it optional in `nuxt.config.ts` sounds like the perfect approach to keep things flexible.

Commit:
feat: introduce overrideDefaults option

volkanakkus commented:
Now, I've added an option called `overrideDefaults?: boolean;` to nuxt.config under the tradingview object, it's `true` by default and existing users will have the same functionality, but if you want to turn off this behavior now you can set: `overrideDefaults: false`, then you'll be able to use your options with the default ones merged. 

I'm merging this PR and I will to add the new option to the documentation, finally I'll publish a new release.

volkanakkus commented:
@aligu7 hey, new patch is released, to use the new option you should update the module to **`v.1.3.4`**,

And for your situation you should use `overrideDefaults: false` in your nuxt.config file. 

Check out more here: 
https://nuxt-tradingview.volkanakkus.com/documentation/getting-started/configuration#overridedefaults
